qa/L0_logging/test.sh

```diff
@@ -0,0 +1,590 @@
+    exit 1
+fi
+
+# Check Default Settings
+rm -f ./curl.out
+set +e
+code=`curl -s -w %{http_code} -o ./curl.out localhost:8000/v2/logging`
+
+# Check if the current setting is returned
+if [ `grep -c "\"log_file\":\""\" ./curl.out` != "1" ]; then
+    echo -e "\n***\n*** Test Failed Incorrect Log File Setting\n***"
+    RET=1
+fi
+if [ `grep -c "\"log_info\":true" ./curl.out` != "1" ]; then
+    echo -e "\n***\n*** Test Failed Incorrect Log Info Setting\n***"
+    RET=1
+fi
+if [ `grep -c "\"log_warnings\":true" ./curl.out` != "1" ]; then
+    echo -e "\n***\n*** Test Failed Incorrect Log Warn Setting\n***"
+    RET=1
+fi
+if [ `grep -c "\"log_errors\":true" ./curl.out` != "1" ]; then
+    echo -e "\n***\n*** Test Failed Incorrect Log Error Setting\n***"
+    RET=1
+fi
+if [ `grep -c "\"log_verbose_level\":0" ./curl.out` != "1" ]; then
+    echo -e "\n***\n*** Test Failed Incorrect Log Verbose Setting\n***"
+    RET=1
+fi
+if [ `grep -c "\"log_format\":\"default\"" ./curl.out` != "1" ]; then
+    echo -e "\n***\n*** Test Failed Incorrect Log Format Setting\n***"
+    RET=1
+fi
+
+for p in {1..10}; do
+    $SIMPLE_HTTP_CLIENT >> client_default.log 2>&1
+    if [ $? -ne 0 ]; then
+        RET=1
+    fi
+
+    $SIMPLE_GRPC_CLIENT >> client_default.log 2>&1
+    if [ $? -ne 0 ]; then
+        RET=1
+    fi
+done
+
+set -e
+
+kill $SERVER_PID
+wait $SERVER_PID
+
+# Test Log File
+SERVER_ARGS="--log-file=log_file.log --model-repository=$MODELSDIR"
+SERVER_LOG="./inference_server_log_file.log"
+run_server
+if [ "$SERVER_PID" == "0" ]; then
+    echo -e "\n***\n*** Failed to start $SERVER\n***"
+    cat $SERVER_LOG
+    exit 1
+fi
+
+rm -f ./curl.out
+set +e
+code=`curl -s -w %{http_code} -o ./curl.out localhost:8000/v2/logging`
+set -e
+
+if [ `grep -c "\"log_file\":\"log_file.log"\" ./curl.out` != "1" ]; then
+    echo -e "\n***\n*** Test Failed Incorrect Log File Setting\n***"
+    RET=1
+fi
+if [ `grep -c "\"log_info\":true" ./curl.out` != "1" ]; then
+    echo -e "\n***\n*** Test Failed Incorrect Log Info Setting\n***"
+    RET=1
+fi
+if [ `grep -c "\"log_warnings\":true" ./curl.out` != "1" ]; then
+    echo -e "\n***\n*** Test Failed Incorrect Log Warn Setting\n***"
+    RET=1
+fi
+if [ `grep -c "\"log_errors\":true" ./curl.out` != "1" ]; then
+    echo -e "\n***\n*** Test Failed Incorrect Log Error Setting\n***"
+    RET=1
+fi
+if [ `grep -c "\"log_verbose_level\":0" ./curl.out` != "1" ]; then
+    echo -e "\n***\n*** Test Failed Incorrect Log Verbose Setting\n***"
+    RET=1
+fi
+if [ `grep -c "\"log_format\":\"default\"" ./curl.out` != "1" ]; then
+    echo -e "\n***\n*** Test Failed Incorrect Log Format Setting\n***"
+    RET=1
+fi
+
+for p in {1..10}; do
+    $SIMPLE_HTTP_CLIENT >> client_test_log_file.log 2>&1
+    if [ $? -ne 0 ]; then
+        RET=1
+    fi
+
+    $SIMPLE_GRPC_CLIENT >> client_test_log_file.log 2>&1
+    if [ $? -ne 0 ]; then
+        RET=1
+    fi
+done
+
+set -e
+
+kill $SERVER_PID
+wait $SERVER_PID
+
```

Comment:
I have a check later that makes sure the file is populated with information, but I agree, it's not the strongest check. I think the cleanest way to approach this is to check in the python backend logging changes first, then modify this script based on a new python model that logs a variety of logs level. This way the script can check against their presence or lack thereof to ensure things are working correctly.

Reply:
I agree that with the Python backend change it would be easier to test if the backend logging is redirected correctly. Besides that, the endpoint logging / core logging redirection should be checked, and this can be done without waiting for the Python backend change. I think you will be able to generate all level of logging by running server in different setting.